lms/templates/dashboard/_dashboard_course_listing.html

```diff
@@ -289,7 +289,28 @@ <h3 class="course-title">
         <%include file="_dashboard_credit_info.html" args="credit_status=credit_status"/>
       % endif
 
-        % if verification_status.get('status') in [VERIFY_STATUS_NEED_TO_VERIFY, VERIFY_STATUS_SUBMITTED, VERIFY_STATUS_RESUBMITTED, VERIFY_STATUS_APPROVED, VERIFY_STATUS_NEED_TO_REVERIFY] and not is_course_blocked:
+      % if show_consent_link and not is_course_blocked:
+        <div class="message message-upsell has-actions is-shown">
+          <div class="wrapper-extended">
+              <p class="message-copy" align="justify">
+                <b class="message-copy-bold">
+                  ${_("Consent to share your data")}
+                </b><br>
+                  ${_("To access this course, you must first consent to share your learning achievements with {enterprise_customer_name}.").format(enterprise_customer_name=enterprise_customer_name)}
+              </p>
+              <div class="action-upgrade-container">
+                <a class="action action-view-consent" href="${course_target}" data-course-key="${enrollment.course_id}">
+                  <span class="wrapper-copy">
+                    <span class="copy" id="view-consent">${_("View Consent")}</span>
+                    <span class="sr">&nbsp;${_(course_overview.display_name_with_default)}</span>
+                  </span>
+                </a>
+              </div>
+          </div>
+        </div>
+      %endif
```

Comment:
It would be good to stick this in a template partial and include it here instead.

Reply:
Good point. Was following the pattern already used in this template where all the cases were inline.

Reply:
Agreed on trying to fit this in it's own html stub. All of the seven current messages under the parent 'messages_list' should be broken out into their own stubs and put into a separate folder, but that is something for another story. 

For here, you can just model it off of the messages above this code block (dashboard_certificate_information and dashboard_credit_info on lines 285 and 289) and name it _dashboard_show_consent.html or something like that.

Reply:
Also, the three messages after this one that you are adding all use 'if not course_blocked' and then have one 'if course_blocked'. Could you see if you can clean that up to be a single conditional? 

`if course_blocked: show course blocked message
else: conditionally show the other three messages
`

Hope that makes sense

Reply:
done, moved to it's own template partial and cleaned up the if statements